### PR TITLE
Make tests compatible with hydrator 3

### PR DESCRIPTION
**Pull request by alexdenvir**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

Builds were failing using latest dependencies on php 7.2+, because Laminas\Hydrator 3.0.0 deprecated the ClassMethods hydrator.

This PR tries to remain backwards compatible by checking if the replacement for ClassMethods is available. This is obviously just a stopgap, as a better solution may be to release a new version that bumps the minimum php version to 7.2 and drop the backwards compatibility altogether

### CODE REVIEW

**Comment by alexdenvir** [comment]
Also, I hope I'm not getting ahead of myself fixing broken builds. I'll hold off if its a nuisance to maintainers

**Comment by michalbundyra** [comment]
@alexdenvir 

> Also, I hope I'm not getting ahead of myself fixing broken builds. I'll hold off if its a nuisance to maintainers

No, I really appreciate your work! :) Thanks a lot 👍

File: test/TableGatewayAbstractFactoryTest.php
Comment (michalbundyra) [inline test/TableGatewayAbstractFactoryTest.php]: It might be better to extract it to separate method, like: `getHydrator()`  (private) so the logic is just in one place.

**Comment by alexdenvir** [comment]
Cool, I'll keep going then :wink:

File: test/TableGatewayAbstractFactoryTest.php
Comment (alexdenvir) [reply test/TableGatewayAbstractFactoryTest.php]: I've added `getClassMethodsHydratorClassName()`, as similar logic was also needed for an assertion

File: test/TableGatewayAbstractFactoryTest.php
Comment (michalbundyra) [inline test/TableGatewayAbstractFactoryTest.php]: please remove above, other than that, LGTM 👍

File: test/TableGatewayAbstractFactoryTest.php
Comment (michalbundyra) [inline test/TableGatewayAbstractFactoryTest.php:252]: would be nice to add the comment before the method, why do we need it - just some clarification for people who are gonna look on the code later on

**Review by michalbundyra — APPROVED** [review]
👍

**Review by samsonasik — APPROVED** [review]

**Comment by michalbundyra** [comment]
Thanks, @alexdenvir!

**Pull request merged**
